flyteidl/protos/flyteidl/core/security.proto

```diff
@@ -45,6 +45,14 @@ message Secret {
     MountType mount_requirement = 4;
 }
 
+message Connection {
+    // The credentials to use for the connection, such as API keys, OAuth2 tokens, etc.
+    map<string, string> secrets = 1;
```

Comment:
Can we also add comment on the format of what the key and what the value is and how they map to what the agent expects